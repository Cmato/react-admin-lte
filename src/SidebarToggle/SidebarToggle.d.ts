import * as React from 'react';

export interface SidebarToggleProps {
  component?: React.ReactType<SidebarToggleProps>;
  onClick?: React.MouseEventHandler<any>;
}

declare const SidebarToggle: React.ComponentType<SidebarToggleProps>;

export default SidebarToggle;
