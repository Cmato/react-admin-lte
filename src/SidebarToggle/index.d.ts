export { default } from './SidebarToggle';
export * from './SidebarToggle';
