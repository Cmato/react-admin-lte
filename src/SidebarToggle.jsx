import React from 'react';

const propTypes = {
  srText: React.PropTypes.string,
};

const SidebarToggle = ({ srText }) => (
  <a href="#" className="sidebar-toggle">
    <span className="sr-only">{srText}</span>
    <span className="icon-bar" />
    <span className="icon-bar" />
    <span className="icon-bar" />
  </a>
);

SidebarToggle.propTypes = propTypes;

export default SidebarToggle;
