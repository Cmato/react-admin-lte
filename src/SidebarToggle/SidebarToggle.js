import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const SidebarToggle = ({
  children,
  className: classNameProp,
  component: Component,
  onClick,
  ...other
}, context) => {
  const className = classNames({
    'sidebar-toggle': true
  }, classNameProp);

  let handleClick;

  if (onClick) {
    handleClick = onClick;
  } else if (context &&
      context.$adminlte_layout &&
      context.$adminlte_layout.onToggle ) {
    handleClick = (e) => { context.$adminlte_layout.onToggle(e) };
  }

  return (
    <Component
      className={className}
      onClick={handleClick}
      {...other}
    >
      {children}
    </Component>
  );
};

SidebarToggle.propTypes = {
  /**
   * The content of the component.
   */
  children: PropTypes.node,
  /**
   * @ignore
   */
  className: PropTypes.string,
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  onClick: PropTypes.func
};

SidebarToggle.defaultProps = {
  component: 'a'
};

SidebarToggle.contextTypes = {
  $adminlte_layout: PropTypes.shape({
    toggleMainSidebar: PropTypes.func
  })
};

export default SidebarToggle;
