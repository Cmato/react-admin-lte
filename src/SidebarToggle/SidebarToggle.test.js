import React from 'react';
import { shallow } from 'enzyme';
import SidebarToggle from './SidebarToggle';

it('renders correctly', () => {
  expect(shallow((
    <SidebarToggle>
      <span className="sr-only">Toggle navigation</span>
    </SidebarToggle>
  ))).toMatchSnapshot();
});

it('toggles through context', () => {
  let received = false;

  const wrapper = shallow(
    <SidebarToggle />,
    {
      context: {
        $adminlte_theme: {
          onChange: () => {
            received = true;
          }
        }
      }
    }
  );

  wrapper.simulate('click');
  expect(received).toEqual(true);
});

it('wires up onClick instead of context', () => {
  let recievedContext = false;
  let receivedClick = false;

  const wrapper = shallow(
    <SidebarToggle
      onClick={() => { receivedClick = true; }}
    />,
    {
      context: {
        $adminlte_theme: {
          onChange: () => {
            receivedContext = true;
          }
        }
      }
    }
  );

  wrapper.simulate('click');
  expect(recievedContext).toEqual(false);
  expect(receivedClick).toEqual(true);
});
