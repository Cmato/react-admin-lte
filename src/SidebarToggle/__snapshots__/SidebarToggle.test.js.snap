// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="sidebar-toggle"
>
  <span
    className="sr-only"
  >
    Toggle navigation
  </span>
</a>
`;
